common/core/core_session.c

```diff
@@ -0,0 +1,482 @@
+ * as defined below, subject to the following condition.
+ *
+ * Without limiting other conditions in the License, the grant of
+ * rights under the License will not include, and the License does not
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+#include "core_session.h"
+
+#include "core.pb.h"
+#include "inheritance_main.h"
+#include "logger.h"
+#include "options.h"
+#include "pb_decode.h"
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+const ecdsa_curve *curve;
+const uint32_t session_key_rotation[2] = {6, 7};
+session_config_t CONFIDENTIAL session = {0};
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief Receives and processes the server's key as part of the session
+ * initialization.
+ *
+ * @param server_message The buffer containing the server's message, which
+ * includes the server's random public key, session age, and device ID.
+ * @return true if the server's key is successfully received and processed.
+ * @return false otherwise.
+ */
+static bool session_receive_server_key(uint8_t *server_message);
```

Comment:
```suggestion
static bool session_receive_server_key(const uint8_t *server_message);
```
the input buffer remains unchanged